src/functional/ffront/foast_passes/type_deduction.py

```diff
@@ -61,6 +71,8 @@ def broadcast_typeinfos(left: TypeInfo, right: TypeInfo) -> Optional[TypeInfo]:
         return None
     if left.is_scalar and right.is_field_type:
         return right
+    elif left.dims and right.dims and len(right.dims) > len(left.dims):
```

Comment:
Just a question: is it guaranteed that the longer list is always at the right?

Reply:
No. If it isn't, then the left typeinfo is returned. So the one with more or same amount of dims.

Reply:
Ok, I see. I misread the code because of the unnecessary `elif` here which I think could be just a simple `if`